src/model.ts

```diff
@@ -292,12 +292,36 @@ export class SpecQueryModel {
         // if the property is a wildcard, return null
         if (isWildcard(encQ[prop])) return null;
 
-        // otherwise, assign the proper to the field def
+        // otherwise, assign the property to the field def
         if (encQ[prop] !== undefined) {
 
           if (!PROPERTY_SUPPORTED_CHANNELS[prop] ||  // all channels support this prop
             PROPERTY_SUPPORTED_CHANNELS[prop][encQ.channel as Channel]) {
             fieldDef[prop] = encQ[prop];
+
+            if (prop === Property.SCALE) {
+              if (isFieldQuery(encQ)) {
+                let field = encQ.field;
+
+                let fieldSchema = this._schema.fieldSchema(field as string);
```

Comment:
- Check if it is ordinal first  and use `schema.domain({field, type})`

- Modify `schema.domain()` to return ordinalDomain if `type` is `"ordinal"` and `ordinalDomain` exists.